storage/tests/SMAIndexSubBlock_unittest.cpp

```diff
@@ -0,0 +1,998 @@
+      generateNumericComparisonPredicate<LongType>(ComparisonID::kEqual, indexed_attr, comparison_lit));
+  std::unique_ptr<SMAPredicate> smapredicate(SMAPredicate::ExtractSMAPredicate(*predicate));
+
+  EXPECT_EQ(indexed_attr, smapredicate->attribute);
+  EXPECT_EQ(ComparisonID::kEqual, smapredicate->comparison);
+  EXPECT_EQ(comparison_lit, smapredicate->literal.getLiteral<std::int64_t>());
+
+  // Test against a comparison which it might flip (less->greater).
+  predicate.reset(generateNumericComparisonPredicate<LongType>(ComparisonID::kLess, indexed_attr, comparison_lit));
+  smapredicate.reset();
+  smapredicate.reset(SMAPredicate::ExtractSMAPredicate(*predicate));
+
+  EXPECT_EQ(indexed_attr, smapredicate->attribute);
+  EXPECT_EQ(ComparisonID::kLess, smapredicate->comparison);
+  EXPECT_EQ(comparison_lit, smapredicate->literal.getLiteral<std::int64_t>());
+
+  // Make a comparison which it must flip.
+  ScalarAttribute *scalar_attribute = new ScalarAttribute(*relation_->getAttributeById(indexed_attr));
+  ScalarLiteral *scalar_literal
+      = new ScalarLiteral(LongType::InstanceNonNullable().makeValue(&comparison_lit),
+                          LongType::InstanceNonNullable());
+  predicate.reset();
+  predicate.reset(new ComparisonPredicate(
+      ComparisonFactory::GetComparison(ComparisonID::kGreater), scalar_literal, scalar_attribute));
+  smapredicate.reset();
+  smapredicate.reset(SMAPredicate::ExtractSMAPredicate(*predicate));
+
+  EXPECT_EQ(indexed_attr, smapredicate->attribute);
+  EXPECT_EQ(ComparisonID::kLess, smapredicate->comparison);
+  EXPECT_EQ(comparison_lit, smapredicate->literal.getLiteral<std::int64_t>());
+}
+
+TEST_F(SMAIndexSubBlockTest, TestGetSelectivity) {
+  // Test to make sure that the selectivity functions behave correctly.
+  // Test with an inline type.
+  std::uint64_t lsmallest = 0, lsmall = 100, lmedium = 1000, llarge = 10000, llargest = 100000;
+  const TypedValue long_smallest = LongType::InstanceNonNullable().makeValue(&lsmallest);
+  const TypedValue long_small = LongType::InstanceNonNullable().makeValue(&lsmall);
+  const TypedValue long_medium = LongType::InstanceNonNullable().makeValue(&lmedium);
+  const TypedValue long_large = LongType::InstanceNonNullable().makeValue(&llarge);
+  const TypedValue long_largest = LongType::InstanceNonNullable().makeValue(&llargest);
+  std::unique_ptr<UncheckedComparator> longs_equal(
+      ComparisonFactory::GetComparison(ComparisonID::kEqual)
+          .makeUncheckedComparatorForTypes(LongType::InstanceNonNullable(), LongType::InstanceNonNullable()));
+  std::unique_ptr<UncheckedComparator> longs_less(
+      ComparisonFactory::GetComparison(ComparisonID::kLess)
+          .makeUncheckedComparatorForTypes(LongType::InstanceNonNullable(), LongType::InstanceNonNullable()));
+
+  // Test with equals.
+  EXPECT_EQ(Selectivity::kNone,
+            sma_internal::getSelectivity(long_smallest, ComparisonID::kEqual,
```

Comment:
We generally write one function argument per line, in case all of them can't fit on one line. If you are refactoring the code, I would strongly suggest to use an IDE to do it, instead of manually changing it. VIM has a plugin called ClangFormat, which can do it for you.